.github/workflows/ci.yml

```diff
@@ -0,0 +1,64 @@
+name: ci
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      - name: Install dev dependencies
+        run: poetry install --no-root --with=dev
+      - name: Run linter
+        run: |
+          source $VENV
+          pre-commit -V
+          pre-commit run --all-files --color=always --show-diff-on-failure
```

Comment:
It is better to only run on the files modified in the commit.

Reply:
fixed